### PR TITLE
MAID-1830 simplify CI scripts

### PR DESCRIPTION
**Pull request by afck**

### CODE REVIEW

**Comment by maidsafe-highfive** [comment]
@afck: no appropriate reviewer found, use r? to override

File: .travis.yml [hunk 1]
```diff
@@ -20,17 +20,22 @@ matrix:
...
+        set -x;
+        cargo fmt -- --write-mode=diff &&
+        cargo rustc --verbose -- --test -Zno-trans &&
+        cargo rustc --verbose --example ci_test -- -Zno-trans &&
+        cargo rustc --verbose --example key_value_store -- -Zno-trans &&
+        cargo rustc --verbose -- -Zno-trans &&
+        cargo test  --verbose --features use-mock-crust
+      );
+    fi
+  - curl -sSL https://github.com/maidsafe/QA/raw/master/Bash%20Scripts/Travis/run_clippy.sh | bash
```
Comment (Viv-Rajkumar) [inline .travis.yml]: this url is in the old format. we can go to [safe_core - travis.yml](https://github.com/maidsafe/safe_core/blob/dev/.travis.yml#L36) format now.

File: .travis.yml [hunk 2]
```diff
@@ -20,17 +20,22 @@ matrix:
...
 before_cache:
-  - curl -sSLO https://github.com/maidsafe/QA/raw/master/Bash%20Scripts/Travis/install_elfutils.sh
-  - . install_elfutils.sh
-after_success:
-  - curl -sSL https://github.com/maidsafe/QA/raw/master/Bash%20Scripts/Travis/after_success.sh | bash
+  - curl -sSL https://github.com/maidsafe/QA/raw/master/Bash%20Scripts/Travis/run_cargo_prune.sh | bash
```
Comment (Viv-Rajkumar) [inline .travis.yml]: this again similar to `safe_core` can just be `cargo prune`

File: .travis.yml [hunk 3]
```diff
@@ -20,17 +20,22 @@ matrix:
...
-  # - cat output.log
+  - if [ "${TRAVIS_RUST_VERSION}" = stable ]; then
+      (
+        set -x;
+        cargo fmt -- --write-mode=diff &&
+        cargo rustc --verbose -- --test -Zno-trans &&
+        cargo rustc --verbose --example ci_test -- -Zno-trans &&
+        cargo rustc --verbose --example key_value_store -- -Zno-trans &&
+        cargo rustc --verbose -- -Zno-trans &&
+        cargo test  --verbose --features use-mock-crust
```
Comment (Viv-Rajkumar) [inline .travis.yml]: I'm guessing running `cargo test` with `use-mock-crust` also includes tests which don't rely on the feature?

Comment (afck) [inline .travis.yml]: Yes, it does. I think the only thing it excludes is `tests/ci.rs`, and we haven't been running that on Travis before this change either.

**Pull request merged**